src/commands/org/open.ts

```diff
@@ -130,48 +123,21 @@ export class OrgOpenCommand extends SfCommand<OrgOpenOutput> {
       throw err;
     }
 
-    const openOptions = flags.browser
-      ? // assertion can be removed once oclif option flag typings are fixed
-        { app: { name: apps[flags.browser as 'chrome' | 'edge' | 'firefox'] } }
-      : {};
+    const openOptions =
+      // assertions could be removed once oclif flag typins are fixed
+      flags.browser ? { app: { name: apps[flags.browser as 'chrome' | 'edge' | 'firefox'] } } : {};
 
-    await utils.openUrl(url, openOptions);
-    return output;
-  }
+    // create a local html file that contains the POST stuff.
+    // for review...there's always an access token, right?
+    const tempFilePath = path.join(tmpdir(), `org:open-${new Date().valueOf()}.html`);
+    await writeFile(tempFilePath, getFileContents(conn.accessToken as string, conn.instanceUrl, retUrl));
+    await utils.openUrl(`file:///${tempFilePath}`, openOptions);
+    // so we don't delete the file while the browser is still using it
+    // open returns when the CP is spawned, but there's not way to know if the browser is still using the file
+    await sleep(1000);
 
-  private async buildFrontdoorUrl(): Promise<string> {
-    await this.org.refreshAuth(); // we need a live accessToken for the frontdoor url
-    const accessToken = this.conn.accessToken;
-    const instanceUrl = this.org.getField<string>(Org.Fields.INSTANCE_URL);
-    const instanceUrlClean = instanceUrl.replace(/\/$/, '');
-    return `${instanceUrlClean}/secur/frontdoor.jsp?sid=${accessToken}`;
-  }
-
-  private async generateFileUrl(file: string): Promise<string> {
-    try {
-      const metadataResolver = new MetadataResolver();
-      const components = metadataResolver.getComponentsFromPath(file);
-      const typeName = components[0]?.type?.name;
-
-      if (typeName === 'FlexiPage') {
-        const flexipage = await this.conn.singleRecordQuery<{ Id: string }>(
-          `SELECT id FROM flexipage WHERE DeveloperName='${path.basename(file, '.flexipage-meta.xml')}'`,
-          { tooling: true }
-        );
-        return `/visualEditor/appBuilder.app?pageId=${flexipage.Id}`;
-      } else if (typeName === 'ApexPage') {
-        return `/apex/${path.basename(file).replace('.page-meta.xml', '').replace('.page', '')}`;
-      } else if (typeName === 'Flow') {
-        return `/builder_platform_interaction/flowBuilder.app?flowId=${await flowFileNameToId(this.conn, file)}`;
-      } else {
-        return 'lightning/setup/FlexiPageList/home';
-      }
-    } catch (error) {
-      if (error instanceof Error && error.name === 'FlowIdNotFoundError') {
-        this.error(error);
-      }
-      return 'lightning/setup/FlexiPageList/home';
-    }
+    await rm(tempFilePath, { force: true });
```

Comment:
if there's an error while deleting file (slow machine, browser still using it after 1s?) `rm` will not retry the operation:
https://nodejs.org/docs/latest-v20.x/api/fs.html#fspromisesrmpath-options

let's set a `maxRetries` just to be safe. I think 3 should be good, will try to break it on QA.

Reply:
good idea.  Looks like the retries needs `recursive:true`